PLD Linux change (Elan Ruusamäe <glen@pld-linux.org>):
use /etc/certs/ca-certificates.crt instead of /etc/pki/tls/certs/ca-bundle.crt

From ae8af24d6d935b2824ad955aa9ea8cc557089c77 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Petr=20P=C3=ADsa=C5=99?= <ppisar@redhat.com>
Date: Fri, 16 Sep 2011 10:33:54 +0200
Subject: [PATCH] Redirect to ca-certificates bundle
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

This patch replaces Mozilla-CA certificate bundle with bundle
delivered by ca-certificates RPM package used as single source of
Mozilla certificate bundle.

See <https://bugzilla.redhat.com/show_bug.cgi?id=738383> for more
details.

Signed-off-by: Petr Písař <ppisar@redhat.com>
---
 MANIFEST          | 1 -
 Makefile.PL       | 8 ++++++++
 lib/Mozilla/CA.pm | 8 +-------
 3 files changed, 9 insertions(+), 8 deletions(-)

diff --git a/MANIFEST b/MANIFEST
index a88847b..6577ede 100644
--- a/MANIFEST
+++ b/MANIFEST
@@ -1,5 +1,4 @@
 lib/Mozilla/CA.pm
-lib/Mozilla/CA/cacert.pem
 Makefile.PL
 MANIFEST			This list of files
 README
diff --git a/Makefile.PL b/Makefile.PL
index 2b10474..57f2f07 100644
--- a/Makefile.PL
+++ b/Makefile.PL
@@ -40,3 +40,11 @@ BEGIN {
         ExtUtils::MakeMaker::WriteMakefile(%arg);
     };
 }
+
+package MY;
+sub MY::libscan {
+    my $name = shift->SUPER::libscan(@_);
+    # Remove private certificate bundle
+    if ($name =~ /cacert.pem\z/) { $name = '' };
+    return $name;
+}
diff --git a/lib/Mozilla/CA.pm b/lib/Mozilla/CA.pm
index 050a494..c9f7dc8 100644
--- a/lib/Mozilla/CA.pm
+++ b/lib/Mozilla/CA.pm
@@ -3,16 +3,10 @@ package Mozilla::CA;
 use strict;
 our $VERSION = '20141217';
 
-use Cwd ();
 use File::Spec ();
-use File::Basename qw(dirname);
 
 sub SSL_ca_file {
-    my $file = File::Spec->catfile(dirname(__FILE__), "CA", "cacert.pem");
-    if (!File::Spec->file_name_is_absolute($file)) {
-	$file = File::Spec->catfile(Cwd::cwd(), $file);
-    }
-    return $file;
+    return File::Spec->catfile('/etc/certs/ca-certificates.crt');
 }
 
 1;
-- 
1.9.3

